packages/core/admin/admin/src/content-manager/components/BlocksEditor/index.js

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+
+import { Flex, Typography } from '@strapi/design-system';
+import { pxToRem } from '@strapi/helper-plugin';
+import PropTypes from 'prop-types';
+import { useIntl } from 'react-intl';
+import { createEditor } from 'slate';
+import { Slate, Editable, withReact, ReactEditor } from 'slate-react';
+import styled from 'styled-components';
+
+const TypographyAsterisk = styled(Typography)`
+  line-height: 0;
+`;
+
+const style = {
+  fontSize: pxToRem(14),
+};
+
+const initialValue = [
+  {
+    type: 'paragraph',
+    children: [{ text: 'A line of text in a paragraph.' }],
+  },
+];
+
+const BlocksEditor = React.forwardRef(({ intlLabel, name, readOnly, required }, ref) => {
+  const { formatMessage } = useIntl();
+  const [editor] = React.useState(() => withReact(createEditor()));
+
+  const label = intlLabel.id
+    ? formatMessage(
+        { id: intlLabel.id, defaultMessage: intlLabel.defaultMessage },
+        { ...intlLabel.values }
+      )
+    : name;
+
+  /** Editable is not able to hold the ref, https://github.com/ianstormtaylor/slate/issues/4082
+   *  so with "useImperativeHandle" we can use ReactEditor methods to expose to the parent above
+   */
+  React.useImperativeHandle(
+    ref,
+    () => ({
+      focus() {
+        ReactEditor.focus(editor);
```

Comment:
Do we actually need autofocus? it seems to me that all attributes would be fighting for focus when opening the edit view

Reply:
I agree, I don't see any focusing issue 🤔 . Did you have a problem @madhurisandbhor? For me it's working without this and the forwardRef. Do you think they could be removed?

Reply:
Yes, customInput needs ref to be defined.Let me check again

Reply:
yes I had console error, you can also check on your side when you do not have forwardRef
<img width="919" alt="Screenshot 2023-09-04 at 15 54 17" src="https://github.com/strapi/strapi/assets/7886244/d4f66420-d55b-4918-931f-690b1b179154">

Reply:
ok I don't have any error but fine with leaving it as is for now